### PR TITLE
Implementation of API calls for cookbook_artifacts end point

### PR DESCRIPTION
**Pull request by mudash**
These changes add functionality for implementation of API calls for `cookbook_artifacts` end point
## Description

`/cookbook_artifacts` end point implementation is needed to fetch cookbooks that are available on Chef Server via Policy Files. `/cookbooks` end point doesn't give access to these cookbooks

## Related Issue

https://app.zenhub.com/workspaces/chef-workstation-5c0aced04b5806bc2bfb9347/issues/chef/go-chef/19

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [X] I have read the **CONTRIBUTING** document.
- [X] I have run the pre-merge tests locally and they pass.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: cookbook_artifacts.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// List lists the Cookbook_Artifacts in the Chef server.
+//  GET /cookbook_artifacts
+func (c *CBAService) List() (data CBAGetResponse, err error) {
+	err = c.client.magicRequestDecoder("GET", "cookbook_artifacts", nil, &data)
+	return
+}
+
+// Get retruns details for a specific cookbook artifact
```
Comment (clintoncwolfe) [inline cookbook_artifacts.go:74]: s/retruns/returns/

File: cookbook_artifacts_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	if err != nil {
+		t.Error(err)
+	}
+
+	if data == nil {
+		t.Fatal("We should have some data")
+	}
+
+	if len(data) != 2 {
+		t.Error("Mismatch in expected policies count. Expected 2, Got: ", len(data))
```
Comment (clintoncwolfe) [inline cookbook_artifacts_test.go:39]: nit: expected cookbook artifacts, not policies

File: cookbook_artifacts.go (hunk 1 above)
Comment (mudash) [reply cookbook_artifacts.go]: Updated. Thanks!

File: cookbook_artifacts_test.go (hunk 2 above)
Comment (mudash) [reply cookbook_artifacts_test.go]: Fixed. Thanks!

File: go.mod [hunk 3]
```diff
@@ -3,14 +3,37 @@ module github.com/chef/go-chef
 go 1.12
 
 require (
+	github.com/aws/aws-sdk-go v1.30.7 // indirect
```
Comment (tyler-ball) [inline go.mod:6]: Most of these dependencies are pulled in for the integration tests. I would love to have those tests pull their own dependencies in with their own `go.mod` but I think we fix that in https://github.com/chef/go-chef/issues/15

Comment (mudash) [reply go.mod:6]: Sounds good. Thanks @tyler-ball

**Pull request merged**